### PR TITLE
add mybatis/hibernate sample

### PR DESCRIPTION
**Pull request by htynkn**
somehow it will not trigger issue which found in 3.0.0

### CODE REVIEW

File: dubbo-samples-boundary-test/dubbo-samples-mybatis/src/test/java/org/apache/dubbo/samples/boundary/mybatis/MybatisServiceIT.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+@ContextConfiguration(locations = "classpath*:spring/mybatis-consumer.xml")
+public class MybatisServiceIT {
+    @Autowired
+    private MybatisService service;
+
+    @Test
+    public void findUser() throws Exception {
+        User user = service.findByUserId(1);
+
+        assertThat(user, not(nullValue()));
```
Comment (kylixs) [inline dubbo-samples-boundary-test/dubbo-samples-mybatis/src/test/java/org/apache/dubbo/samples/boundary/mybatis/MybatisServiceIT.java:25]: It is recommended to check user.getName():
```
  Assert.assertEquals(user.getName(), "apache dubbo");
```

**Review by kylixs — COMMENTED** [review]
It is recommended to move the `dubbo-samples-mybatis` project to the root directory

File: dubbo-samples-boundary-test/dubbo-samples-mybatis/src/test/java/org/apache/dubbo/samples/boundary/mybatis/MybatisServiceIT.java (hunk 1 above)
Comment (htynkn) [reply dubbo-samples-boundary-test/dubbo-samples-mybatis/src/test/java/org/apache/dubbo/samples/boundary/mybatis/MybatisServiceIT.java:26]: yeah, you are right. fixed via b996041

**Comment by htynkn** [comment]
> It is recommended to move the `dubbo-samples-mybatis` project to the root directory



> It is recommended to move the `dubbo-samples-mybatis` project to the root directory

the purpose for this mybatis project is to verify dubbo works well with mybatis. (I'm trying to add a hibernate demo too).
this is more like a test to verify function.

It doesn't use any new dubbo feature and it's also not a suggested best practice. that's why I put it in `dubbo-samples-boundary-test` folder

**Comment by kylixs** [comment]
> It doesn't use any new dubbo feature and it's also not a suggested best practice. that's why I put it in dubbo-samples-boundary-test folder 

If this is the purpose, I prefer to change `dubbo-samples-boundary-test` to `integration-tests`

File: dubbo-samples-boundary-test/dubbo-samples-hibernate/case-versions.conf [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+
+# Supported component versions of the test case
+
+# Spring app
+dubbo.version=2.7.*, 3.*
+spring.version=4.*
```
Comment (kylixs) [inline dubbo-samples-boundary-test/dubbo-samples-hibernate/case-versions.conf:24]: Not only test Spring 4.x, this example should have no special Spring version requirements and can use the wildcard *
```
spring.version=*
```

Comment (htynkn) [reply dubbo-samples-boundary-test/dubbo-samples-hibernate/case-versions.conf:24]: > Not only test Spring 4.x, this example should have no special Spring version requirements and can use the wildcard *
> 
> ```
> spring.version=*
> ```



> > It doesn't use any new dubbo feature and it's also not a suggested best practice. that's why I put it in dubbo-samples-boundary-test folder
> 
> If this is the purpose, I prefer to change `dubbo-samples-boundary-test` to `integration-tests`

I'm ok with any name changing, @AlbumenJ what do you think

Comment (htynkn) [reply dubbo-samples-boundary-test/dubbo-samples-hibernate/case-versions.conf:24]: It does have version requirement for spring. for both mybatis and hibernate.
they have different version requirement. if we need change spring version, we need modify hibernate version too.

Comment (AlbumenJ) [reply dubbo-samples-boundary-test/dubbo-samples-hibernate/case-versions.conf:24]: > > Not only test Spring 4.x, this example should have no special Spring version requirements and can use the wildcard *
> > ```
> > spring.version=*
> > ```
> 
> > > It doesn't use any new dubbo feature and it's also not a suggested best practice. that's why I put it in dubbo-samples-boundary-test folder
> > 
> > 
> > If this is the purpose, I prefer to change `dubbo-samples-boundary-test` to `integration-tests`
> 
> I'm ok with any name changing, @AlbumenJ what do you think

Looks fine. BTW, it is more important to add some description of this module.

**Comment by kylixs** [comment]
Use spring-boot is fine, but need change `case-versions.conf` to:

//spring-boot 1.x
```
dubbo.version=2.7*, 3.*
spring-boot.version=1.*
```

//spring-boot 2.x
```
dubbo.version=2.7*, 3.*
spring-boot.version=2.*
```

//both spring-boot 1.x/ 2.x
```
dubbo.version=2.7*, 3.*
spring-boot.version=1.*, 2.*
```

**Comment by htynkn** [comment]
> Use spring-boot is fine, but need change `case-versions.conf` to:
> 
> //spring-boot 1.x
> 
> ```
> dubbo.version=2.7*, 3.*
> spring-boot.version=1.*
> ```
> 
> //spring-boot 2.x
> 
> ```
> dubbo.version=2.7*, 3.*
> spring-boot.version=2.*
> ```
> 
> //both spring-boot 1.x/ 2.x
> 
> ```
> dubbo.version=2.7*, 3.*
> spring-boot.version=1.*, 2.*
> ```

done.

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**